recipes/msix/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.33.0"
+
+
+class MsixConan(ConanFile):
+    name = "msix"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/msix-packaging"
+    description = "An SDK for creating MSIX packages"
+    topics = ("msix", "sdk", "packaging", "conan-recipe")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_lib": ["crypt32", "openssl"],
+        "pack": [True, False],
+        "skip_bundles": [True, False],
+        "use_external_zlib": [True, False],
+        "use_validation_parser": [True, False],
+        "xml_parser": ["applexml", "javaxml", "msxml6", "xerces"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "crypto_lib": "openssl",
+        "pack": False,
+        "skip_bundles": False,
+        "use_external_zlib": True,
+        "use_validation_parser": False,
+        "xml_parser": "xerces"
+    }
+
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self.settings.os == "Android":
+            self._cmake.definitions["AOSP"] = True
+        if self.settings.os == "Linux":
+            self._cmake.definitions["LINUX"] = True
+        if self.settings.os == "Macos":
+            self._cmake.definitions["MACOS"] = True
+        self._cmake.definitions["CRYPTO_LIB"] = self.options.crypto_lib
+        self._cmake.definitions["MSIX_PACK"] = self.options.pack
+        self._cmake.definitions["MSIX_SAMPLES"] = False
+        self._cmake.definitions["MSIX_TESTS"] = False
+        self._cmake.definitions["SKIP_BUNDLES"] = self.options.skip_bundles
+        self._cmake.definitions["USE_EXTERNAL_ZLIB"] = self.options.use_external_zlib
+        self._cmake.definitions["USE_SHARED_ZLIB"] = self.options["zlib"].shared
+        self._cmake.definitions["USE_VALIDATION_PARSER"] = self.options.use_validation_parser
+        self._cmake.definitions["XML_PARSER"] = self.options.xml_parser
+        self._cmake.configure()
+        return self._cmake
+
+    def validate(self):
```

Comment:
this recipe has lot of logic based on dependencies options.

Tries to reorder methods by order of execution:
config_options => configure => requirements => validate => package_id => build_requirements => build => package => package_info

It will be easier to see caveats.

Reply:
[Done](https://github.com/conan-io/conan-center-index/pull/6475/commits/2fe4bea8d5be2b27ef1dd4de1aa6eb272cdbada5)

Reply:
About moving  `_configure_cmake`, I'm not sure it's a good idea to mix "private" and "public" methods, but if it is a standard for CCI I will fix the recipe.

Reply:
We "mix" all the time.
It's more "natural" to us to have the methods in order of execution.
It saves us a lot of scrolling in order to find methods.

Reply:
The `validate` method was moved to match the order of execution in the above mentioned commit, I told only about `_configure_cmake`, I don't think that the term `order of execution` can be applied to it since we can invoke it wherever we want.